### ISSUE TITLE
Building Nuke 7.6 with Swift 4.2 with Carthage causes problems

### ISSUE THREAD

**Issue by willemfrishert** [kean/Nuke#221]
I'm using Carthage to pull Nuke update and use Xcode 10.1 with Swift 4.2.1 to build locally.
`github "kean/Nuke" ~> 7.5.2`

After Nuke 7.6 got released, I'm no longer able to build Nuke with Xcode 10.1 and Swift 4.2.1.
In more detail, the problem appears after
```
"Upgrade project to recommended settings"
d3e586ea1ea5977327feef0eaf18039f9d96bca2
```

I have two questions:
1. I was under the impression from the release notes Nuke 7.6 that only Swift 4.0 and 4.1 were dropped but Swift 4.2 remains active?
2. Should removing support for iOS 9 and Swift 4.0 and 4.1 not automatically result in a change to a major version number?

**Comment by kean** [comment]
Nuke 7.6 does support Swift 4.2. I'm not entirely sure how to handle this situation with Carthage. If there is no way to do that I would either recommend upgrading to Swift 5 or installing Nuke 7.5.2.

> Should removing support for iOS 9 and Swift 4.0 and 4.1 not automatically result in a change to a major version number?

It's debatable. I personally reserve major versions for major changes in Nuke itself. I recently added a [requirements list](https://github.com/kean/Nuke#requirements) in README.md which lists all recent Nuke version to make it clear which versions you can use.

**Comment by willemfrishert** [comment]
So to follow (and for anyone also reading this thread).
- I only have Xcode 10.1 (with Swift 4.2) installed
- I do not have Xcode 10.2 installed (no trace of Swift 5.0)

I've tested the following step:
1. Download the source code directly from the release page (remove any need for Carthage);
2. Open Xcode 10.1 and use the Nuke.xcconfig as is with SWIFT_VERSION=5.0;
3. Change SWIFT_VERSION=4.2 in the xcconfig file

At step 2.: Xcode 10.1 immediate fails with the message: `error: SWIFT_VERSION '5.0' is unsupported, supported versions are: 3.0, 4.0, 4.2. (in target 'Nuke')`
At step 3.: Xcode 10.1 compiles Nuke 7.6 successfully with SWIFT_VERSION=4.2

So in conclusion it's the SWIFT_VERSION=5.0 and it's Xcode (or Xcodebuild) does not like. 

@kean in the release notes you posted, it states that Xcode 10.1 with Swift 4.2 would be supported but I'm not sure how to implement support for 4.2 and 5.0 when it depends on SWIFT_VERSION and 5.0 is only introducing the ABI compatibility from that moment forward.

For the time being, I've taken your suggestion to limit our setup to Nuke 7.5.2 and wait with adopting 7.6 after we've migrated to Xcode 10.2 with Swift 5.0 support.

**Comment by kean** [comment]
If you find a solution, please post it if possible. I tried briefly googling it and couldn't find anything related.

> For the time being, I've taken your suggestion to limit our setup to Nuke 7.5.2 and wait with adopting Nuke 7.6 after we've migrated to Xcode 10.2 with Swift 5.0 support.

This would probably be the best option. Nuke 7.6 doesn't contain any other changes other than Swift 5 support anyway. And there are no known bugs so there is really no need to update.
